streams/shared/src/main/scala/zio/stream/ZStream.scala

```diff
@@ -2646,6 +2646,18 @@ object ZStream extends Serializable {
   final def apply[R, E, A](pull: ZManaged[R, E, Pull[R, E, A]]): ZStream[R, E, A] =
     new ZStream[R, E, A](ZStream.Structure.Iterator(pull))
 
+  /**
+   * Creates a stream from a value received from the accessed environment.
+   */
+  final def access[R, A](f: R => A): ZStream[R, Nothing, A] =
```

Comment:
We should use the partial type application trick. This works but requires the user to specify both type parameters (e.g. `ZStream.accessM[Random, Int](_.random.nextInt)`). If we have `access` return a new `AccessPartiallyAppliedType` like in ZIO then we can allow this to be written as `ZStream.accessM[Random](_.random.nextInt)` and the return type will be inferred. This is more convenient for the user and still preserves the type information. Sorry didn't think of this earlier.

Reply:
yeah - that's what I mentioned in the issue description, but I didn't know any solution to that. thanks, I'll try to do it as suggested :)